dask/array/overlap.py

```diff
@@ -520,6 +556,11 @@ def map_overlap(x, func, depth, boundary=None, trim=True, **kwargs):
     depth2 = coerce_depth(x.ndim, depth)
     boundary2 = coerce_boundary(x.ndim, boundary)
 
+    for i in range(x.ndim):
+        if isinstance(depth2[i],tuple):
+            if boundary2[i] != 'none':
+                raise NotImplementedError
```

Comment:
Can we add some explanation in the exception about what exactly is not implemented?

Reply:
yes of course, thanks